src/web/js-interop.md

```diff
@@ -22,3 +22,77 @@ For help using the `js` package, see the following:
 [js-api]: {{site.pub-api}}/js
 [sass]: {{site.pub-pkg}}/sass
 
+## Future-proof JS interop
+
+Dart's JS interop story is currently evolving. 
+Many of the features that enable future JS interop,
+and support of all three compilers,
+are ready to use (Dart version 2.19). 
+To ease the transition to future JS interop,
+you can refactor your code to conform to the
+syntax and semantics of the new features now.
+
+{{site.alert.note}}
+  Future-proofing is for those *very* eager to
+  prepare their code for [Dart 3][], or to try
+  integrating with [`dart2wasm`][] before Dart 3.
+  None of this necessarily works *performantly* on
+  `dart2wasm` yet, but refactoring now means you
+  won't need to later, if that's important to you.
+{{site.alert.end}}
+
+The following sections are the set of features
+expected to work across compilers for JS interop,
+per library.
+
+*Requirements:*
+* Dart SDK constraint: `>= 2.19`
+* [`package:js`][] constraint: `>= 0.6.6`
+
+[Dart 3]: https://medium.com/dartlang/the-road-to-dart-3-afdd580fbefa
+[`dart2wasm`]: https://github.com/dart-lang/sdk/blob/main/pkg/dart2wasm/dart2wasm.md#running-dart2wasm
+[`package:js`]: https://pub.dev/packages/js
+
+### `package:js`
+
+If you're interested in interop that will work with
+`dart2wasm` when Dart 3 is released, you can implement
+[static interop][] using the `package:js` class `@staticInterop`.
+Static interop will be continually eveloving until Dart 3.
+
+The set of features for future static interop currently includes:
+* `@staticInterop` interfaces
+  * External factory constructors with and without `@anonymous`
+  * External `static` class members
+  * External non-`static` extension members on a `@staticInterop`
+  class (fields, getters, setters, methods)
+  * Non-external extension members on a `@staticInterop` class
+* Top-level external members
+* [`@JSExport`][] for mocking and exports
+
+[static interop]: https://github.com/dart-lang/sdk/tree/main/pkg/js#staticinterop
+[`@JSExport`]: https://github.com/dart-lang/sdk/tree/main/pkg/js#jsexport-and-js_utilcreatedartexport
+
+### `dart:js_util`
+
+If you're interested in refactoring your [`js_util`][]
+interop code in preparation for Dart 3, the following
+list of features for the library currently work:
```

Comment:
"currently work" in the context of what? Work at all? Across web compilers including `dart2wasm`?

\cc @srujzs

Reply:
I'm not sure if it's worth highlighting the list of APIs here. This list is practically the entire set of APIs exposed in `dart:js_util`.

Long term our recommendation is to use inline-classes with static JSInterop whenever that's possible. And only use js_util when it's absolutely necessary. For example, we have a bug and you need a workaround, or there is something that can't be expressed with static interop (possibly an omission we'd like to fix as well).

It may come up that some migrations are easier if js_util is available. For example, preexisting uses of `dart:js` (now deprecated) can be replaced by a combination of static interop and dart:js_util. The former is what we recommend, but it may be easier to automate migrations using the latter :shrug:

Reply:
Ok, so I'll remove the list of APIs, and just describe the general long-term goal of only using js_util for edge cases that static interop isn't designed to handle?

Reply:
@sigmundch One more question about this: are you basically saying that `js_util` provides lower-level functionality that can be configured in a more customized way than static interop? So you're keeping it around just in case some unknown cases go unaccounted for in static interop's design?

Reply:
Exactly. But we expect that to be rare. For context, in the past it was helpful to address some of the limitations of non-static-interop `package:js` that couldn't be done before we added static-interop.

One thing we want to make clear in our messaging is that we recommend static-interop, that static-interop is more declarative, and as a result more likely to be optimized and more likely to perform better. We may optimize js_util patterns, but that is not a guarantee we will promise in our backends.